code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py

```diff
@@ -137,11 +116,43 @@ def __init__(self, volt_max : int|None = None, volt_min : int|None = None,
             curr_max (int): [description]
             curr_min (int): [description]
         '''
+        if ((volt_max is None and #pylint: disable=too-many-boolean-expressions
+             (volt_min is not None or curr_max is not None or curr_min is not None))
+            or (volt_min is None and
+            (curr_max is not None or curr_min is not None or volt_max is not None))
+            or (curr_max is None and
+            (curr_min is not None or volt_max is not None or volt_min is not None))
+            or (curr_min is None and
+            (volt_max is not None or volt_min is not None or curr_max is not None))):
+            log.error("Invalid power range")
+            raise ValueError("Invalid power range")
+        if (volt_max is not None and volt_min is not None and volt_max < volt_min):
```

Comment:
Check only range is correct. Previously None type has been checked

Reply:
Done